lte/gateway/python/integ_tests/s1aptests/test_ics_timer_expiry_with_mme_restart.py

```diff
@@ -0,0 +1,135 @@
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper(
+            stateless_mode=MagmadUtil.stateless_cmds.ENABLE
+        )
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def test_ics_timer_expiry_with_mme_restart(self):
+        """
+        The test case validates the functionality of ICS timer expiry while UE
+        is in idle mode after mme restart
+        Step 1: Attach UE to network
+        Step 2: Let the UE goes into Idle mode
+        Step 3: Send Service Request to move the ECM state to connected mode
+        Step 4: Send an idication to s1ap module to drop ICS Request message
+                So s1ap shall not send ICS Response message to mme
+        Step 5: While ICS timer is running at mme, restart mme service
+        Step 6: On mme recovery, it is expected  that ICS timer has lapsed and
+                Send Ue Context Release Command to s1ap
+        """
+        self._s1ap_wrapper.configUEDevice(1)
+        req = self._s1ap_wrapper.ue_req
+        ue_id = req.ue_id
+        print(
+            "************************* Running End to End attach for UE id ",
+            ue_id,
+        )
+        # Attach
+        self._s1ap_wrapper._s1_util.attach(
+            ue_id,
+            s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+            s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+            s1ap_types.ueAttachAccept_t,
+        )
+
+        # Wait on EMM Information from MME
+        self._s1ap_wrapper._s1_util.receive_emm_info()
+        time.sleep(0.5)
+
+        print(
+            "************************* Sending UE context release request ",
+            "for UE id ",
+            ue_id,
+        )
+        # Send UE context release request to move UE to idle mode
+        req = s1ap_types.ueCntxtRelReq_t()
+        req.ue_Id = ue_id
+        req.cause.causeVal = gpp_types.CauseRadioNetwork.USER_INACTIVITY.value
+        self._s1ap_wrapper.s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_CNTXT_REL_REQUEST, req
+        )
+        response = self._s1ap_wrapper.s1_util.get_response()
+        self.assertEqual(
+            response.msg_type, s1ap_types.tfwCmd.UE_CTX_REL_IND.value
+        )
+
+        print("*** Sending indication to drop Initial Context Setup Req ***")
+        drop_init_ctxt_setup_req = s1ap_types.UeDropInitCtxtSetup()
+        drop_init_ctxt_setup_req.ue_Id = ue_id
+        drop_init_ctxt_setup_req.flag = 1
+        # Timer to release UE context at s1ap tester
+        drop_init_ctxt_setup_req.tmrVal = 2000
+        self._s1ap_wrapper._s1_util.issue_cmd(
+            s1ap_types.tfwCmd.UE_SET_DROP_ICS, drop_init_ctxt_setup_req
+        )
+
+        print(
+            "************************* Sending Service request for UE id ",
+            ue_id,
+        )
+        # Send service request to reconnect UE
+        req = s1ap_types.ueserviceReq_t()
+        req.ue_Id = ue_id
+        req.ueMtmsi = s1ap_types.ueMtmsi_t()
+        req.ueMtmsi.pres = False
```

Comment:
Why is the presence flag marked as False if we are setting the value in line 98?

Reply:
If present flag is set to true, then mTmsi is provided from test script, otherwise mTmsi is taken from UE context of ue_app.